Branch-SDK/Branch-SDK/BNCServerRequestQueue.m

```diff
@@ -157,9 +157,21 @@ - (BOOL)containsClose {
 
 #pragma mark - Private method
 
-- (void)persist {
-    NSUserDefaults *defaults = [NSUserDefaults standardUserDefaults];
+- (void)persistEventually {
+    if (!self.writeTimer.valid) {
+        self.writeTimer = [NSTimer scheduledTimerWithTimeInterval:BATCH_WRITE_TIMEOUT target:self selector:@selector(persistToDisk) userInfo:nil repeats:NO];
+    }
+}
+
+- (void)persistImmediately {
```

Comment:
This is a public method, and is called in a couple places within the `Branch.m` file, like when the app goes to background.